From ccd29124d0d2276a3071c0418c14dec188cd3727 Mon Sep 17 00:00:00 2001
From: Mark Renouf <mrenouf@google.com>
Date: Mon, 8 Jul 2024 14:27:37 -0400
Subject: [PATCH] Prevent Sharing when FRP enforcement is in effect

ADB command to trigger sharing:

```
adb shell 'am start -a android.intent.action.CHOOSER --eu android.intent.extra.INTENT "intent:#Intent;action=android.intent.action.SEND;type=text/plain;S.android.intent.extra.TEXT=Shared%20text;end"'
```

Bug: 327645387
Test: Manually, trigger FRP, use adb to trigger sharing
Flag: EXEMPT bugfix
(cherry picked from commit 94d1e1e4e9539437ec0549b7bf22999054b92f1f)
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:a530903d6bf9c0ec35ac008f2e4ac8772a6da40b)
Merged-In: I4116ecbd3534b0391a5b468ed4cd18f2d4b3ae0a
Change-Id: I4116ecbd3534b0391a5b468ed4cd18f2d4b3ae0a
---
 java/src/com/android/intentresolver/ChooserActivity.java | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/java/src/com/android/intentresolver/ChooserActivity.java b/java/src/com/android/intentresolver/ChooserActivity.java
index 63ac643..a206e8a 100644
--- a/java/src/com/android/intentresolver/ChooserActivity.java
+++ b/java/src/com/android/intentresolver/ChooserActivity.java
@@ -57,6 +57,7 @@ import android.os.SystemClock;
 import android.os.UserHandle;
 import android.os.UserManager;
 import android.os.storage.StorageManager;
+import android.provider.Settings;
 import android.service.chooser.ChooserTarget;
 import android.util.Log;
 import android.util.Slog;
@@ -229,6 +230,12 @@ public class ChooserActivity extends ResolverActivity implements
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
+        if (Settings.Global.getInt(getContentResolver(), Settings.Global.SECURE_FRP_MODE, 0) == 1) {
+            Log.e(TAG, "Sharing disabled due to active FRP lock.");
+            super.onCreate(savedInstanceState);
+            finish();
+            return;
+        }
         Tracer.INSTANCE.markLaunched();
         final long intentReceivedTime = System.currentTimeMillis();
         mLatencyTracker.onActionStart(ACTION_LOAD_SHARE_SHEET);
-- 
2.46.0.469.g59c65b2a67-goog

